### PR TITLE
Handle unknown opcode in input

### PR DESCRIPTION
**Pull request by spectranaut**
### Resolves

https://github.com/LLK/scratch-vm/issues/1682

### Proposed Changes

Fix import error when block has an input with an unknown opcode.

### Reason for Changes

#1254 skips unknown opcodes when importing blocks from sb2 projects. However, that PR missed the case where an input is undefined. This PR fixes projects that were failing to load because of that issue.

### Test Coverage

This is my first issue, I'm not sure how to test other than manually with this project: https://llk.github.io/scratch-gui/develop/#116599254

I added a unit test called "unknown-opcode-as-reporter-block.js"

### How to test this

1. In Scratch 2.0, drag out a block with an input
2. Put a block with unknown opcodes in the input. Examples are the "picoboard" extension blocks, such as "sensor value"
3. Import the project into 3.0
4. The project should be able to load and there will be a default value (shadow) instead of the input. There should be warnings about skipped opcodes in the console.

### CODE REVIEW

File: src/serialization/sb2.js [hunk 1]
```diff
@@ -843,6 +843,7 @@ const parseBlock = function (sb2block, addBroadcastMsg, getVariableId, extension
     for (let i = 0; i < blockMetadata.argMap.length; i++) {
         const expectedArg = blockMetadata.argMap[i];
         const providedArg = sb2block[i + 1]; // (i = 0 is opcode)
+
```
Comment (kchadha) [inline src/serialization/sb2.js]: Nit: stray newline

File: src/serialization/sb2.js [hunk 2]
```diff
@@ -866,32 +867,35 @@ const parseBlock = function (sb2block, addBroadcastMsg, getVariableId, extension
...
+                    innerBlocks = parsedBlockDesc[0] ? [parsedBlockDesc[0]] : [];
                     // Update commentIndex
                     commentIndex = parsedBlockDesc[1];
                 }
                 parseState.expectedArg = parentExpectedArg;
-                // Check if innerBlocks is an empty list.
-                // This indicates that all the inner blocks from the sb2 have
+
+                // Obscures any shadow.
+                shadowObscured = true;
```
Comment (kchadha) [inline src/serialization/sb2.js]: You should move this line below the `if (innerBlocks.length > 0)` block below since it's not being used till later

**Review by kchadha — CHANGES_REQUESTED** [review]
A few minor comments, and then it's good to go!

**Comment by spectranaut** [comment]
addressed your feedback, @kchadha ! :)

**Review by kchadha — APPROVED** [review]
LGTM!

**Pull request merged**